From 27403437e065a257153ddd8d7edcf2d5c9cc11e8 Mon Sep 17 00:00:00 2001
From: Shikun Ding <shikun.ding@intel.com>
Date: Thu, 21 Apr 2022 12:28:43 +0800
Subject: [PATCH] Reset max9295 reg to succeed in streaming y12i

When max9295_init_settings() is invoked at the max9295 probe
stage or streamed y8/y8i before running y12i, the 0x31E register
is overridden by a non-default value.  Any change on the 0x31E
register for enabling y8/y8i may cause y12i not to work since
the change on that reg is not required in y12i.

Reset the 0x31E reg to enable y12i.

Signed-off-by: Shikun Ding <shikun.ding@intel.com>
---
 drivers/media/i2c/max9295.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/i2c/max9295.c b/drivers/media/i2c/max9295.c
index 87f127677..eb3b4df22 100644
--- a/drivers/media/i2c/max9295.c
+++ b/drivers/media/i2c/max9295.c
@@ -552,6 +552,10 @@ static struct reg_pair map_pipe_z_y12i_control[] = {
 	{0x0318, 0x64}, // Pipe Z pulls Y12I (DT 0x24)
 	{0x030D, 0x04}, // Pipe Z pulls VC2
 	{0x030E, 0x00},
+	/* Reset reg 0x031E since it's very likely to modified in Y8/Y8I
+	 * before run Y12I. For Y12I, this reg not required to be set.
+	 */
+	{0x031E, 0x18},
 	{0x0112, 0x0E}, // LIM_HEART Pipe Z: Disabled
 };
 
-- 
2.17.1

